packages/plugin-core/src/commands/GoToSiblingCommand.ts

```diff
@@ -33,65 +34,207 @@ export class GoToSiblingCommand extends BasicCommand<
   }
 
   async execute(opts: CommandOpts) {
+    // Context for error messages
     const ctx = "GoToSiblingCommand";
-    const maybeTextEditor = VSCodeUtils.getActiveTextEditor();
-    if (!maybeTextEditor) {
+    // Get text editor
+    const textEditor = VSCodeUtils.getActiveTextEditor();
+    if (!textEditor) {
       window.showErrorMessage("You need to be in a note to use this command");
       return {
         msg: "no_editor" as const,
       };
     }
-    let value = "";
-    value = path.basename(maybeTextEditor.document.uri.fsPath, ".md");
-    let respNodes: NoteProps[] = [];
+    // Get fname of the active note
+    const fname = path.basename(textEditor.document.uri.fsPath, ".md");
+    // Get workspace
+    const workspace = ExtensionProvider.getDWorkspace();
+    // Get the active note
+    const note = await this.getActiveNote(workspace.engine, fname);
+    if (!note) throw new Error(`${ctx}: ${UNKNOWN_ERROR_MSG}`);
+    // Get the sibling note
+    let siblingNote: NoteProps;
+    // If the active note is a journal note, get the sibling note based on the chronological order
+    if (await this.canBeHandledAsJournalNote(note, workspace.engine)) {
+      const { sibling, msg } = await this.getSiblingForJournalNote(
+        workspace.engine.notes,
+        note,
+        opts.direction
+      );
+      if (msg) return { msg };
+      siblingNote = sibling!;
+    } else {
+      const { sibling, msg } = this.getSibling(
+        workspace,
+        note,
+        opts.direction,
+        ctx
+      );
+      if (msg) return { msg };
+      siblingNote = sibling!;
+    }
+    // Get path to the vault
+    const vpath = vault2Path({
+      vault: siblingNote.vault,
+      wsRoot: workspace.wsRoot,
+    });
+    // Open the sibling note
+    await VSCodeUtils.openFileInEditor(
```

Comment:
you can use [openNote](https://github.com/dendronhq/dendron/blob/enchance/wrap-month-journal-nav/packages/plugin-core/src/WSUtilsV2.ts#L261:L261) method